src/Compilers/Core/CodeAnalysisTest/Text/TextChangeTests.cs

```diff
@@ -880,6 +902,164 @@ public void TestMergeChanges_NoMiddleMan()
             Assert.Equal("o World", changes[0].NewText);
         }
 
+        [Fact]
+        [WorkItem(47234, "https://github.com/dotnet/roslyn/issues/47234")]
+        public void DebuggerDisplay()
+        {
+            Assert.Equal("new TextChange(new TextSpan(0, 0), null)", default(TextChange).GetDebuggerDisplay());
+            Assert.Equal("new TextChange(new TextSpan(0, 1), \"abc\")", new TextChange(new TextSpan(0, 1), "abc").GetDebuggerDisplay());
+            Assert.Equal("new TextChange(new TextSpan(0, 1), (NewLength = 10))", new TextChange(new TextSpan(0, 1), "0123456789").GetDebuggerDisplay());
+        }
+
+        [Fact]
+        [WorkItem(47234, "https://github.com/dotnet/roslyn/issues/47234")]
+        public void Fuzz()
+        {
+            var random = new Random();
+
+            // Adjust upper bound as needed to generate a simpler reproducer for an error scenario
+            var originalText = SourceText.From(string.Join("", Enumerable.Range(0, random.Next(10))));
+
+            foreach (var iteration in Enumerable.Range(0, 100000))
+            {
+                var editedLength = originalText.Length;
+                ArrayBuilder<TextChange> oldChangesBuilder = ArrayBuilder<TextChange>.GetInstance();
+
+                const int maxEditLength = 3;
+                const int maxSkipLength = 2;
+                // generate sequence of "old edits" which meet invariants
+                for (int i = 0; i < originalText.Length; i += 1 + random.Next(maxSkipLength))
```

Comment:
Strictly speaking, I think we should be able to delete the `1 +` here, and let valid inputs have adjacent spans, but I wasn't able to solve a particular crashing bug related to it.

Reply:
Solved it :smile: